packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -845,24 +1027,23 @@ class IOSDeviceLogReader extends DeviceLogReader {
   /// is true.
   final List<String> _streamFlutterMessages = <String>[];
 
-  /// When using both `idevicesyslog` and `ios-deploy`, exclude logs with the
```

Comment:
I may need some help understanding how log reader works - can you explain what are `idevicesyslog` and `ios-deploy` and how are they different.

Reply:
So there are three ways of getting logs from the device: `idevicesyslog`, `ios-deploy`, and by getting logs from the Dart VM (it's called UnifiedLogging in the code). `idevicesyslog` uses syslog to get the logs, `ios-deploy` uses lldb I believe, UnifiedLogging uses the Dart VM. 

`ios-deploy` does not work with iOS 17. `idevicesyslog` does not work with wireless devices. Dart VM is not as comprehensive of logs and is only used when no other log sources work (like iOS 17 wireless devices).

So we use a combination of these log sources in different situations. We listen to the log streams for the various sources and send all the messages to a single StreamController, which is used by various other processes (like to print the logs out in verbose and get the Dart VM).
https://github.com/flutter/flutter/blob/d75a01283b5c2efd3d71d3f33c971251cb6db51f/packages/flutter_tools/lib/src/ios/devices.dart#L1009-L1024

Reply:
Gotcha. Great explanation. Can you add some comments here for future readers?